include/deviceconfig.h

```diff
@@ -112,7 +112,7 @@ class DeviceConfig : public IJSONSerializable
 
     DeviceConfig();
 
-    virtual bool SerializeToJSON(JsonObject& jsonObject) override

```

Comment:
I did not know it was preferred to remove the virtual decoration in overrides. As you can tell! Now I know. :-)

Reply:
Oh, I know totally how it happens as I've been there. If you have code (or a programmer) that predates C++11, you have the "virtual Foo" that slots into the over the abstract code. Then C++ 11 comes along and you spray an "override" on it because you've bitten by that ONE virtual method that had a slightly mismatched signature and wouldn't get called by the abstract class and took you a day to fine, so you condition yourself for the override. But we all (observed in a large group of very smart people) forget to take the virtual's out.

To your question, I'm very good with one-liners in vim, perl, sed, and beating stones together. The process of doing THIS one admittedly frustrated me enough to capture the pio build output into a 'compile_commands.json' that I then used to build a CMakeLists.txt that worked just well enough to let me load it into CLion (free for open source maintainer) which happens to be very good at this kind of that. That spooled me up for the CL that followed this one (It does spelling and grammar check, too - and no, I didn't grind the boot on every nit) so while that CLion build isn't good enough to build/run/debug, it's good enough to EDIT in a way that conveniently brings clang-tidy (whose modernizer could have been used for THIS one, but wasn't) output right into the editor with convenient little "fix" light bulbs next to the code under the microscope.

So i started off by mining my own ore and refining it into a toolsmith and end the end had a decent blacksmithing shop.

My former employer had Very Good tools for doing this kind of thing.